src/interaction/EfInteraction.py

```
@@ -143,8 +163,18 @@ def send_key_down(self, key, activate=True):
+        # 后台模式下：按下时前置游戏（pynput 只投递到前台窗口），松开时恢复原窗口
+        if self._background_mode():
+            if self._background_key_hold_count == 0:
+                self._key_prev_hwnd = win32gui.GetForegroundWindow()
+            self._background_key_hold_count += 1
         if activate:
-            active_and_send_mouse_delta(self._game_hwnd(), only_activate=True)
+            hwnd = self._game_hwnd()
+            was_foreground = win32gui.GetForegroundWindow() == hwnd
+            active_and_send_mouse_delta(hwnd, only_activate=True)
+            if not was_foreground:
+                # 等待窗口真正置顶后再按键，避免按键投递到旧前台窗口
```

Ruff (0.16.2)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)